pkg/providers/imagefamily/agentbakerbootstrap/provisionclientbootstrap.go

```diff
@@ -0,0 +1,239 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package agentbakerbootstrap
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/karpenter-provider-azure/pkg/apis/v1alpha2"
+	"github.com/Azure/karpenter-provider-azure/pkg/operator/options"
+	"github.com/Azure/karpenter-provider-azure/pkg/provisionclients/client"
+	"github.com/Azure/karpenter-provider-azure/pkg/provisionclients/client/operations"
+	"github.com/Azure/karpenter-provider-azure/pkg/provisionclients/models"
+	"github.com/Azure/karpenter-provider-azure/pkg/utils"
+
+	core "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+
+	corev1beta1 "sigs.k8s.io/karpenter/pkg/apis/v1beta1"
+	"sigs.k8s.io/karpenter/pkg/cloudprovider"
+
+	httptransport "github.com/go-openapi/runtime/client"
+	"github.com/go-openapi/strfmt"
+	"github.com/samber/lo"
+)
+
+type ProvisionClientBootstrap struct {
+	ClusterName                    string
+	KubeletConfig                  *corev1beta1.KubeletConfiguration
+	Taints                         []core.Taint      `hash:"set"`
+	StartupTaints                  []core.Taint      `hash:"set"`
+	Labels                         map[string]string `hash:"set"`
+	SubnetID                       string
+	Arch                           string
+	SubscriptionID                 string
+	ClusterResourceGroup           string
+	ResourceGroup                  string
+	KubeletClientTLSBootstrapToken string
+	KubernetesVersion              string
+	ImageDistro                    string
+	IsWindows                      bool
+	InstanceType                   *cloudprovider.InstanceType
+	StorageProfile                 string
+	ImageFamily                    string
+}
+
```

Comment:
Most of these options are shared between self contained and agentbaker bootstrap. Can we share those values with struct embedding rather than copy pasting in both places?